pypeit/calibrations.py

```diff
@@ -560,40 +562,55 @@ def get_flats(self):
+#            # tslits_dict to reflect new slit edges
+#            if self.par['flatfield']['tweak_slits']:
+#                # flatfield updates slits directly and only alters the
+#                # *_tweak set. This updates the MasterEdges
+#                msgs.info('Using slit boundary tweaks from IllumFlat and updated tilts image')
+##                self.tslits_dict = self.flatField.tslits_dict
+#                self.tilts_dict = self.flatField.tilts_dict
 
             # Save to Masters
             if self.save_masters:
                 self.flatField.save()
 
-                # If we tweaked the slits update the master files for tilts and slits
-                # TODO: These should be saved separately
+                # If slits were tweaked by the slit illumination
+                # profile, re-write them so that the tweaked slits are
+                # included.
                 if self.par['flatfield']['tweak_slits']:
-                    msgs.info('Updating MasterTrace and MasterTilts using tweaked slit boundaries')
-                    self.edges.update_using_tslits_dict(self.flatField.tslits_dict)
-                    self.edges.save()
-                    # Write the final_tilts using the new slit boundaries to the MasterTilts file
-                    self.waveTilts.final_tilts = self.flatField.tilts_dict['tilts']
-                    self.waveTilts.tilts_dict = self.flatField.tilts_dict
-                    self.waveTilts.save()
+                    self.slits.to_master()
+
+#                # If we tweaked the slits update the master files for tilts and slits
+#                # TODO: These should be saved separately
+#                if self.par['flatfield']['tweak_slits']:
+#                    msgs.info('Updating MasterTrace and MasterTilts using tweaked slit boundaries')
+#                    # flatfield updates slits directly and only alters
+#                    # the *_tweak set. This updates the MasterEdges
+#                    # file with the new data from the flat-field slit
+#                    # tweaks.
+#                    # TODO: Make SlitTraceSet a masterframe?
+#                    self.edges.update_slits(self.slits)
+#                    self.edges.save()
+
+                    # TODO: Tilts are unchanged, right?
+#                    # Write the final_tilts using the new slit boundaries to the MasterTilts file
+#                    self.waveTilts.final_tilts = self.flatField.tilts_dict['tilts']
+#                    self.waveTilts.tilts_dict = self.flatField.tilts_dict
+#                    self.waveTilts.save()
 
         # 4) If either of the two flats are still None, use unity
         # everywhere and print out a warning
         # TODO: These will barf if self.tilts_dict['tilts'] isn't
         # defined.
         if self.mspixelflat is None:
-            self.mspixelflat = np.ones_like(self.tilts_dict['tilts'])
+#            self.mspixelflat = np.ones_like(self.tilts_dict['tilts'])
```

Comment:
No.  I think we need to get out of the happen of allocating arrays when we can just set them to None.  It avoids the time and memory overhead.  It's not critical in this instance because we only apply the flat once per detector.  The pixel flat was allowed to be None in the image processing, so I was banking on that to take care of this.  But following the execution path, I had to add a couple of lines to `ProcessRawImage.process`.  Do we have a test that runs the image processing where the pixel flat is set to None?